### PR TITLE
Room Settings 2: Integrated `RoomSettingsGetter` svc in `RoomSettingsController#show` API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Synchronize room settings with rooms configurations.

This PR completes **2.** 
--
~~1. Create `RoomSettingsGetter` service that can take a room and return its settings as a `Hash: :name => :value` making sense of each setting and its config.~~
~~2. Integrate the service in `RoomSettingsController#show`.~~
3. Extend the `RoomSettings` and `RoomSettingsRow` UIs.

### User story [Room Settings show]:
---
1. A user authenticates.
2. A user creates and accesses a room and its settings.

[If the setting config is "true" for the room setting]:
4. User will have the setting enabled and user cannot disable it.
>NOTE: The API will set the setting value as if it was enabled, the front end only will disable the setting for editing but will not change its value.

[If the setting config is "false" for the room setting]:
3. User will not see the setting.
>NOTE: The frontend will not mount the setting component to the DOM.

[If the setting config is "optional" for the room setting]:
3. User will have their setting with its registered status 'enabled|disbaled'.
5. User can alter the status of the setting and have proper feedbacks.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3699

File: spec/controllers/room_settings_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+      get :show, params: { friendly_id: room.friendly_id }
+      expect(response).to have_http_status(:ok)
+      expect(JSON.parse(response.body)['data']).to eq({ 'setting' => 'value' })
+    end
+  end
+
+  describe 'room_settings_controller#update' do
+    skip 'TODO' do
+      # TODO: Add `room_settings_controller#update` tests.
```
Comment (KH-Amir-TN) [inline spec/controllers/room_settings_controller_spec.rb:23]: This test is written in #3714

**Comment by KH-Amir-TN** [comment]
# REBASED ON e5d4db1421e6f936b473c5a21437283d1f3f1c11

**Review by hadicheaito1 — DISMISSED** [review]

**Review by farhatahmad — APPROVED** [review]

**Comment by KH-Amir-TN** [comment]
### @farhatahmad PR is ready 👨‍🍳!

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**